### PR TITLE
ci: add helpful comment to PR when lint fails

### PR DESCRIPTION
**Pull request by nbolton**
Fixes: https://github.com/deskflow/deskflow/issues/7527

TODO: Instead of commit ammdend and rebase (which GitHub doesn't really want you to do because of infinite loop bugs), we will add a comment to the PR instructing the author how to fix the problem.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-cmake_minimum_required(VERSION 3.5)
+  cmake_minimum_required(VERSION 3.5)
```
Comment (sithlord48) [inline CMakeLists.txt:17]: Please remove this unneeded change

**Review by sithlord48 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (nbolton) [reply CMakeLists.txt]: This is to trigger the linter. I'll let you know when the PR is out of draft for you to review.

Comment (sithlord48) [reply CMakeLists.txt]: Looks like its not pushing back to the repo

Comment (nbolton) [reply CMakeLists.txt]: Oh I've been having fun with that one! 

It does, but I keep pushing up the lint error to re-trigger it.

See #7529 and #7528 for my previous mistakes; turns out if you --amend to fetch depth 1, it pushes changes to all files. I figured out the solution is to add:
```
        with:
          ref: ${{ github.event.pull_request.head.ref || github.ref }}
          fetch-depth: 2
```
to
```
      - name: Checkout
        uses: actions/checkout@v4
```

Comment (sithlord48) [reply CMakeLists.txt]: why not depth of 0 ?

Comment (nbolton) [reply CMakeLists.txt]: I don't think that depth is needed; `0` brings in all the tags which I don't need here. Depth of `2` works well.

The problem I have now is that GitHub workflows can't be triggered by GitHub (e.g. if GitHub does `git push`), so once the linter makes it's fixes, the workflow doesn't re-run and gets stuck... not quite sure where to go from here. I'll sleep on it.

image

**Comment by sithlord48** [comment]
Moved to discussion https://github.com/deskflow/deskflow/discussions/7536

**Pull request closed without merging**